.github/workflows/pr-scan.yml

```diff
@@ -1,8 +1,8 @@
 name: PR Security Scan
 on:
-  pull_request:
-    types: [opened, synchronize, reopened]
-  workflow_dispatch:
+  pull_request_target:
+    branches: ['**']
+
```

Comment:
Let's limit the access of the GITHUB_TOKEN by specifying [permissions](https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token).